ta/sims/ta_sims.c

```diff
@@ -136,3 +136,41 @@ TEE_Result sims_get_counter(void *session_context, uint32_t param_types,
 
 	return TEE_SUCCESS;
 }
+
+TEE_Result sims_entry_panic(void *session_context, uint32_t param_types,
+			    TEE_Param params[4])
+{
+	uint32_t ret_orig;
+	TEE_Result res = TEE_SUCCESS;
+	struct sims_session *ctx = (struct sims_session *)session_context;
+
+	if (param_types != TEE_PARAM_TYPES(TEE_PARAM_TYPE_MEMREF_INPUT,
+					   TEE_PARAM_TYPE_NONE,
+					   TEE_PARAM_TYPE_NONE,
+					   TEE_PARAM_TYPE_NONE))
+		return TEE_ERROR_BAD_PARAMETERS;
+
+	if (params[0].memref.buffer && params[0].memref.size) {
+		res = sims_open_ta_session(session_context,
+					   param_types, params);
+		if (res != TEE_SUCCESS)
+			return res;
+
+		params[0].memref.buffer = NULL;
+		params[0].memref.size = 0;
```

Comment:
minor: maybe also `param_types = 0;`.

Reply:
I would rather not reset `param_types`, this is leading to a false positive result in the use case when it's triggered a panic from a TA session. The returned result is `TEEC_SUCCESS` (0) instead of `TEEC_ERROR_TARGET_DEAD` (0xFFFF3024) in file regression_1000.c at line 1692.
Do you need more details about the case?

Reply:
True that TA panic invocation command expects param0 is a input memref.
Fine as is.